src/Symfony/Component/Emoji/Resources/bin/build.php

```diff
@@ -131,105 +128,73 @@ public static function buildRules(array $emojisCodePoints): Generator
+        [$map, $reverse] = self::createRules($maps, true);
 
-        return ['emoji-github' => $maps, 'github-emoji' => array_flip($maps)];
+        return ['emoji-github' => $map, 'github-emoji' => $reverse];
     }
 
     public static function buildGitlabRules(array $emojisCodePoints): iterable
     {
         $emojis = json_decode((new Filesystem())->readFile(__DIR__.'/vendor/gitlab-emojis.json'), true, flags: JSON_THROW_ON_ERROR);
-
-        $ignored = [];
         $maps = [];
 
         foreach ($emojis as $emojiItem) {
             $emojiCodePoints = strtolower($emojiItem['unicode']);
+
             if (!array_key_exists($emojiCodePoints, $emojisCodePoints)) {
-                $ignored[] = [
-                    'emojiCodePoints' => $emojiCodePoints,
-                    'name' => $emojiItem['name'],
-                ];
                 continue;
             }
             $emoji = $emojisCodePoints[$emojiCodePoints];
-            if (!self::testEmoji($emoji, 'gitlab', $emojiCodePoints)) {
-                continue;
-            }
-            $codePointsCount = mb_strlen($emoji);
-            $maps[$codePointsCount][$emoji] = $emojiItem['shortname'];
+            $maps[mb_strlen($emoji)][$emojiItem['shortname']] = $emoji;
         }
 
-        $maps = self::createRules($maps);
+        [$map, $reverse] = self::createRules($maps, true);
 
-        return ['emoji-gitlab' => $maps, 'gitlab-emoji' => array_flip($maps)];
+        return ['emoji-gitlab' => $map, 'gitlab-emoji' => $reverse];
     }
 
     public static function buildSlackRules(array $emojisCodePoints): iterable
     {
         $emojis = json_decode((new Filesystem())->readFile(__DIR__.'/vendor/slack-emojis.json'), true, flags: JSON_THROW_ON_ERROR);
-
-        $ignored = [];
-        $emojiSlackMaps = [];
-        $slackEmojiMaps = [];
+        $maps = [];
 
         foreach ($emojis as $data) {
             $emojiCodePoints = str_replace('-', ' ', strtolower($data['unified']));
-            $shortCode = $data['short_name'];
-            $shortCodes = $data['short_names'];
-            $shortCodes = array_map(fn ($v) => ":$v:", $shortCodes);
 
             if (!array_key_exists($emojiCodePoints, $emojisCodePoints)) {
-                $ignored[] = [
-                    'emojiCodePoints' => $emojiCodePoints,
-                    'shortCode' => $shortCode,
-                ];
                 continue;
             }
             $emoji = $emojisCodePoints[$emojiCodePoints];
-            if (!self::testEmoji($emoji, 'slack', $emojiCodePoints)) {
-                continue;
-            }
-            $codePointsCount = mb_strlen($emoji);
-            $emojiSlackMaps[$codePointsCount][$emoji] = ":$shortCode:";
-            foreach ($shortCodes as $short_name) {
-                $slackEmojiMaps[$codePointsCount][$short_name] = $emoji;
+            $codePointsCount = mb_strlen($emoji) << 1;
+            $maps[($codePointsCount << 1) + 1][":{$data['short_name']}:"] = $emoji;
```

Comment:
why this `+ 1` here ?

Reply:
the preferred name should have higher priority than its aliases